packages/components/tests/dummy/app/templates/components/modal.hbs

```diff
@@ -0,0 +1,669 @@
+        <code class="dummy-code">...attributes</code>
+        spreading.</p>
+    </dd>
+  </dl>
+</section>
+
+<section>
+  <h3 class="dummy-h3" id="how-to-use"><a href="#how-to-use" class="dummy-link-section">§</a> How to use</h3>
+
+  <h4 class="dummy-h4">Basic use</h4>
+  <p class="dummy-paragraph">The simplest way to invoke a modal is by doing something like this:</p>
+  {{! prettier-ignore-start }}
+  {{! template-lint-disable no-unbalanced-curlies }}
+  <CodeBlock
+    @language="markup"
+    @code='
+      <Hds::Button @text="Open basic modal" @color="secondary" \{{on "click" (fn this.activateModal "basicModalActive")}} />
+
+      \{{#if this.basicModalActive}}
+        <Hds::Modal id="basic-modal" @onClose=\{{fn this.deactivateModal "basicModalActive"}} as |M|>
+          <M.Header>
+            Modal title
+          </M.Header>
+          <M.Body>
+            <p class="hds-typography-body-300 hds-foreground-primary">Modal content</p>
+          </M.Body>
+          <M.Footer>
+            <Hds::Button type="submit" @text="Confirm" \{{on "click" (fn this.deactivateModal "basicModalActive")}} />
+          </M.Footer>
+        </Hds::Modal>
+      \{{/if}}
+    '
+  />
+  {{! template-lint-enable no-unbalanced-curlies }}
+  {{! prettier-ignore-end }}
+  <p class="dummy-paragraph">Renders to:</p>
+  <Hds::Button @text="Open basic modal" @color="secondary" {{on "click" (fn this.activateModal "basicModalActive")}} />
+
+  {{#if this.basicModalActive}}
+    <Hds::Modal id="basic-modal" @onClose={{fn this.deactivateModal "basicModalActive"}} as |M|>
+      <M.Header>
+        Modal title
+      </M.Header>
+      <M.Body>
+        <p class="hds-typography-body-300 hds-foreground-primary">Modal content</p>
+      </M.Body>
+      <M.Footer>
+        <Hds::Button type="submit" @text="Confirm" {{on "click" (fn this.deactivateModal "basicModalActive")}} />
+      </M.Footer>
+    </Hds::Modal>
+  {{/if}}
+  <p class="dummy-banner dummy-banner--info dummy-paragraph"><FlightIcon @name="info" />
+    Note: When a modal dialog is open with the keyboard, the focus is automatically set to the first focusable element
+    inside the dialog, which is the "Dismiss" button. The action of this button has no effect on the system, so focusing
+    on it helps prevent users from accidentally confirming.
+  </p>
+
+  <h4 class="dummy-h4">Form within a modal dialog</h4>
+  {{! prettier-ignore-start }}
+  {{! template-lint-disable no-unbalanced-curlies }}
+  <CodeBlock
+    @language="markup"
+    @code='
+      <Hds::Button @text="Open form modal" @color="secondary" \{{on "click" (fn this.activateModal "formModalActive")}} />
+
+      \{{#if this.formModalActive}}
+        <Hds::Modal id="form-modal" @onClose=\{{fn this.deactivateModal "formModalActive"}} as |M|>
+          <M.Header>
+            Why do you want to leave the beta?
+          </M.Header>
+          <M.Body>
+            <Hds::Form::Select::Field autofocus @width="100%" as |F|>
```

Comment:
[question] how did we get around the template-linter's no-autofocus rule?

Reply:
Sorry for the delay on this – we do disable the linter for this specific line.